lemur/models.py

```diff
@@ -0,0 +1,312 @@
+                                                    'Lab.id')))
+association_table_role_power = db.Table('association_role_power',
+                                        db.Column('role_name', db.String(64),
+                                                  db.ForeignKey('Role.name')),
+                                        db.Column('power_id', db.String(64),
+                                                  db.ForeignKey(
+                                                    'Power.id')))
+
+
+# set up class to track the date and time information for an object
+class DateTimeInfo(object):
+    created_on = db.Column(db.DateTime, default=datetime.now)
+    updated_on = db.Column(db.DateTime, default=datetime.now,
+                           onupdate=datetime.now)
+
+
+# Store general info of a lab
+class Lab(db.Model):
+    __tablename__ = 'Lab'
+    id = db.Column(db.String(128), nullable=False, unique=True,
+                   primary_key=True)
+    name = db.Column(db.String(128), nullable=False)
+    class_name = db.Column(db.String(128), nullable=False)
+    prof_name = db.Column(db.String(128), nullable=False)
+    # lab description contained in description variable
+    description = db.Column(db.String(4096))
+    status = db.Column(db.Enum('Activated', 'Downloadable', 'Unactivated',
+                               name='status'))
+
+    # one-to-many: a lab can have multiple experiments
+    experiments = db.relationship('Experiment', back_populates='lab',
+                                  cascade="all, delete, delete-orphan")
+    # Many-to-Many: a class can have multiple labs and a lab can have multiple
+    # classes
+    classes = db.relationship("Class", secondary=association_table_class_lab,
+                              back_populates="labs")
+    # Many-to-Many: a user can have multiple labs and a lab can have multiple
+    # users
+    users = db.relationship("User", secondary=association_table_user_lab,
+                            back_populates="labs")
+
+    def __repr__(self):
+        # Representation of class object in string format
+        tpl = ('Lab<id: {id}, name: {name}'
+               ', class_name: {class_name}, prof_name: {prof_name}'
+               ', description: {description}, status: {status}'
+               ', experiments: {experiments}, classes: {classes}'
+               ', users: {users}>')
+        formatted = tpl.format(id=self.id, name=self.name,
+                               class_name=self.class_name,
+                               prof_name=self.prof_name,
+                               description=self.description,
+                               status=self.status,
+                               experiments=[e.name for e in self.experiments],
+                               classes=[c.name for c in self.classes],
+                               users=[u.name for u in self.users])
+        return formatted
+
+
+# Store experiments' info of a lab
+class Experiment(db.Model):
```

Comment:
I think I responded there too, basically saying that I agree with your decisions given this model. My bad for not understanding the app :)

I'd add comments with the correct names for now, just to help the reader along a little.